src/main/scala/cognite/spark/v1/AssetHierarchyBuilder.scala

```diff
@@ -118,112 +196,173 @@ class AssetHierarchyBuilder(config: RelationConfig)(val sqlContext: SQLContext)
       IO.unit
     }
 
-  def getCdfRootAndTree(sourceRootExternalId: CogniteExternalId): IO[(Option[Asset], List[Asset])] = {
-    val cdfTree = client.assets
-      .filter(AssetsFilter(rootIds = Some(Seq(sourceRootExternalId))))
-      .compile
-      .toList
+  def fetchCdfRoot(sourceRootExternalId: String): IO[Option[Asset]] =
+    client.assets
+      .retrieveByExternalId(sourceRootExternalId)
+      .map(Some(_).asInstanceOf[Option[Asset]])
```

Comment:
Use a type ascription here `Some(_): Option[Asset]` to avoid a runtime cast, or use `_.some` from cats